### PR TITLE
Fix using LM locks, SharedMemory left almost untouched

### PR DESCRIPTION
**Pull request by AlexPeshkoff**

### CODE REVIEW

**Comment by AlexPeshkoff** [comment]
Another fix for #7255

File: src/jrd/tpc.cpp [hunk 1]
```diff
@@ -146,24 +149,38 @@ void TipCache::finalizeTpc(thread_db* tdbb)
...
+			if (nmSnap.hasData())
+				SharedMemoryBase::unlinkFile(nmSnap.c_str());
+			if (nmHdr.hasData())
+				SharedMemoryBase::unlinkFile(nmHdr.c_str());
+		}
+		else
+			tdbb->tdbb_status_vector->init();
+	}
+
+	LCK_release(tdbb, m_lock);
```
Comment (hvlad) [inline src/jrd/tpc.cpp:183]: If LCK_lock() in line 172 failed, here we'll get bugcheck in LM

Comment (hvlad) [reply src/jrd/tpc.cpp:183]: Actually, not a bugcheck (due to additional check in LCK), but anyway - this is not fully correct

File: src/jrd/tpc.cpp [hunk 2]
```diff
@@ -375,14 +405,30 @@ TipCache::StatusBlockData::~StatusBlockData()
...
+	if (fName.hasData())
+	{
+		if (LCK_lock(tdbb, &existenceLock, LCK_EX, LCK_NO_WAIT))
+		{
+			SharedMemoryBase::unlinkFile(fName.c_str());
+			LCK_release(tdbb, &existenceLock);
+		}
+		else
+			tdbb->tdbb_status_vector->init();
+	}
```
Comment (hvlad) [inline src/jrd/tpc.cpp:431]: Here same optimization as in `TipCache::finalizeTpc()` could be applied - don't call `LCK_release` before `LCK_lock`.
I'm not insist on it, though.

**Pull request merged**